docs/DeveloperGuide.md

```diff
@@ -6,8 +6,70 @@
 
 ## Design & implementation
 
+### Architecture
+
+### UI Component
+
+# UserCommand feature
+
+User can type command to do things on the Map. 
+
+If the user type the command, `parseCommand` function will be called. After that, the original input will be
+analyzed by `analyseCommand` function to see if it matches a kind of `CommandType`. Then, we will return new
+Command back to the main based on the command type. The Final step is to call the `execute` function. If the command
+is the type of `fightCommand`, we will call the execute function with one parameter `Scanner`. For all other
+conditions, we will call the execute function with no parameter.
+
+![dummy test](/picture/Command.svg)
+
+### Map Component
+
+The API of this component is defined in BaseMap.java.
+
+Each map instance consists of a `currentMap` which stores a 2-dimensional array of characters which represents the 
+printed map for the player. All maps will come with a given `height` and `width`, all of these attributes are inherited
+from the AMap abstract class. Currently, the `FirstMap` and `BattleInterface` classes
+extend AMap. `FirstMap` is the first map displayed upon entering the game and it displays the position of the player. 
+The `BattleInterface` is the map displayed when the player interacts with an `interactable`. 
+
+
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Interacting with Environment Component
+
+The API of this component is defined in InteractingCommand.java
+
+![Interaction UML](../picture/Interaction.png)
+
+This component happens when the user chooses to key the interact command. <br>
+Here is how it works:
+1.  When the command is passed by the user, the ```CalculaChroniclesOfTheAlgorithmicKingdom``` object calls the ```Parser``` object
+    to parse the command to call the respective commands. Here, the ```InteractingCommand``` object is created.
+2. The ```CalculaChroniclesOfTheAlgorithmicKingdom``` component then calls the execute() method in ```InteractingCommand```.
+3. It executes the method and creates other objects like ```Enemy``` and ```ShopKeaper``` components which are responsible for the 
+entity classes in the game and also ```BattleInterface``` and ```ShopMap``` which are responsible for displaying these entities among other things.
+4. The ```BattleInterface``` or ```ShopKeeper``` will then read from a .txt file to store their displays by creating the ```FileReader``` object and 
+running their respective methods.
+5. These displays, textboxes and player status will then be subsequently printed by the ```CalculaChroniclesOfTheAlgorithmicKingdom``` object.
+
+### Battling Component
```

Comment:
There are very many classes and packages, the diagram is very useful to help explain. Perhaps some of the helper classes and packages can also be explained in words.